lms/djangoapps/bulk_email/tasks.py

```diff
@@ -403,11 +405,36 @@ def _get_source_address(course_id, course_title):
     # For the email address, get the course.  Then make sure that it can be used
     # in an email address, by substituting a '_' anywhere a non-(ascii, period, or dash)
     # character appears.
-    from_addr = u'"{0}" Course Staff <{1}-{2}>'.format(
-        course_title_no_quotes,
-        re.sub(r"[^\w.-]", '_', course_id.course),
-        settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
-    )
+    course_name = re.sub(r"[^\w.-]", '_', course_id.course)
+
+    # Translators: Bulk email from address e.g. ("Physics 101" Course Staff)
+    course_staff_title = _('"{course_title}" Course Staff')
+
+    from_addr_format = course_staff_title + u' <{course_name}-{from_email}>'
+
+    def format_address(course_title_no_quotes):
+        """
+        Partial function for formatting the from_addr. Since
+        `course_title_no_quotes` may be truncated to make sure the returned
+        string has fewer than 320 characters, we define this function to make
+        it easy to determine quickly what the max length is for
+        `course_title_no_quotes`.
+        """
+        return from_addr_format.format(
+            course_title=course_title_no_quotes,
+            course_name=course_name,
+            from_email=settings.BULK_EMAIL_DEFAULT_FROM_EMAIL,
+        )
+
+    from_addr = format_address(course_title_no_quotes)
+
+    # if it's longer than 320 characters, reformat, but with a truncated
+    # course title
+    if len(from_addr) >= 320:
+        max_course_title_length = 320 - len(format_address("")) - 5
+        truncated_course_title = course_title_no_quotes[:max_course_title_length] + u"..."
```

Comment:
Some more thoughts:

We should use an ellipses character instead of 3 dots: https://www.drupal.org/node/847608

Pages 25-32 of this presentation provide some more information on the complexities of splitting unicode strings: http://hoytech.github.io/truncate-presentation/. And this library implements segmentation based on [Unicode Text Segmentation standard](http://unicode.org/reports/tr29/): http://uniseg-python.readthedocs.org/en/latest/. I understand course titles may be simple, but if the library does do the job, it will be useful in other places too.